src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -550,15 +552,15 @@ protected async Task FireOnCompleted()
                     }
                     catch (Exception ex)
                     {
-                        Log.ApplicationError(_logger, ((IHttpConnectionFeature)this).ConnectionId, TraceIdentifier, ex);
+                        Log.ApplicationError(_logger, ((IHttpConnectionFeature)this).ConnectionId, TraceIdentifier!, ex); // TODO: Can TraceIdentifier be null?
                     }
                 }
             }
         }
 
         public void SetBadRequestState(BadHttpRequestException ex)
         {
-            Log.ConnectionBadRequest(_logger, RequestConnectionId, ex);
+            Log.ConnectionBadRequest(_logger, RequestConnectionId!, ex); // TODO: Can RequestConnectionId be null?
```

Comment:
It looks like RequestConnectionId is only set when [accessing the ConnectionId](https://github.com/dotnet/aspnetcore/blob/master/src/Servers/IIS/IIS/src/Core/IISHttpContext.IHttpConnectionFeature.cs#L103). Should we be using ConnectionId instead?

Reply:
Probably best fix here is to access the RequestConnectionId via the explicit interface implementation.

Reply:
Yeah that seems good for both this and the trace identifier.